vm/src/exceptions.rs

```diff
@@ -519,6 +520,18 @@ impl PyBaseException {
         let cls = zelf.class();
         format!("{}({})", cls.name(), repr_args.iter().format(", "))
     }
+
+    #[pymethod(magic)]
+    fn reduce(
+        zelf: PyRef<Self>,
+        _vm: &VirtualMachine,
+    ) -> (PyTypeRef, PyTupleRef, Option<PyDictRef>) {
+        (
+            zelf.class().clone(),
+            zelf.args(),
+            zelf.as_object().dict(),
+        )
```

Comment:
Last field must not be included in tuple if it is `None`. This leads to print empty set like this
```python
>>>>> BaseException("test").__reduce__()
(<class 'BaseException'>, ('test',), {})
```
It must be printed like this
```python
>>>>> BaseException("test").__reduce__()
(<class 'BaseException'>, ('test',))
```
It is good to refer this cpython code 😊
```c
static PyObject *
BaseException_reduce(PyBaseExceptionObject *self, PyObject *Py_UNUSED(ignored))
{
    if (self->args && self->dict)
        return PyTuple_Pack(3, Py_TYPE(self), self->args, self->dict);
    else
        return PyTuple_Pack(2, Py_TYPE(self), self->args);
}
```

Reply:
Resolved in e2af440.